shell/common/api/electron_api_clipboard.cc

```diff
@@ -180,8 +180,16 @@ void Clipboard::WriteBookmark(const base::string16& title,
 
 gfx::Image Clipboard::ReadImage(gin_helper::Arguments* args) {
   ui::Clipboard* clipboard = ui::Clipboard::GetForCurrentThread();
-  SkBitmap bitmap = clipboard->ReadImage(GetClipboardBuffer(args));
-  return gfx::Image::CreateFrom1xBitmap(bitmap);
+  base::Optional<gfx::Image> image;
+  clipboard->ReadImage(
+      GetClipboardBuffer(args),
+      base::Bind(
+          [](base::Optional<gfx::Image>* image, const SkBitmap& result) {
+            image->emplace(gfx::Image::CreateFrom1xBitmap(result));
+          },
+          &image));
+  DCHECK(image.has_value());
```

Comment:
Would it be possible to prevent against a future explosion here where this gets converted to actually-async and we don't hit this DCHECK in our tests? Right now if the `DCHECK` is compiled out and this executes actually asynchronously, this will write to a random place in memory.

Reply:
I don't see a way without creating the image object on heap and using `scoped_refptr`. We can do that or convert the api to promise asap on master.